backend/django/core/templates/projects/update/permissions.html

```diff
@@ -81,5 +81,37 @@ <h3>Instructions</h3>
     $('tr.formset_row_perm').last().children().eq(1).find('ul.errorlist li').empty();
   }
 });
+
+// Deduplication of profiles on permission page
+function dedupPermProfiles() {
+  var selected = [];
+  $("select[id$='profile']").each(function() {
+      if (this.value != "") selected.push(this.value);
+  })
+  $("select[id$='profile']").each(function() {
+      $(this).find("option:not(:selected)").each(function() {
+        // previously unselected option is now selected
+        if ($.inArray(this.value, selected) > -1) {
+          $(this).attr("disabled", "disabled");
+        } else {
+          $(this).removeAttr("disabled");
+        }
+    })
+  })
+}
+
+$("#permSelect").ready(function() {
+  dedupPermProfiles();
+})
+
+// New profile-permission row is added
+$("#permSelect").on("DOMNodeInserted", "tr", function() {
+  dedupPermProfiles();
+})
+
+// Any other change to permission form
+$("#permSelect").on("change", function() {
+  dedupPermProfiles();
+})
```

Comment:
If this JS is the same in multiple files, you could create a single JS file and import it as a script into both templates.

Perhaps, add a dir in `STATICFILES_DIRS` for scripts and place your script there. Then you can use it a [Django Static File](https://docs.djangoproject.com/en/4.0/howto/static-files/)